csharp/src/Ice/Communicator.cs

```diff
@@ -1068,54 +1027,53 @@ async Task PerformDisposeAsync()
 
                 if (_adminAdapter != null)
                 {
-                    Debug.Assert(_adminIdentity != null);
-                    return _adminAdapter.CreateProxy(_adminIdentity.Value, IObjectPrx.Factory);
+                    return _adminAdapter.CreateProxy(_adminIdentity!.Value, IObjectPrx.Factory);
```

Comment:
We should assert here that _adminEnabled or move the check for !_adminEnabled before this

Reply:
Added assert.